diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2024-10-03 21:15:50.000000000 +0200
+++ b/CMakeLists.txt	2025-02-21 20:26:43.071191152 +0100
@@ -60,9 +60,9 @@
     target_compile_definitions(${PROJECT_NAME} INTERFACE SZ3_DEBUG_TIMINGS=0)
 endif ()
 
-pkg_search_module(ZSTD IMPORTED_TARGET libzstd)
-if (ZSTD_FOUND AND NOT SZ3_USE_BUNDLED_ZSTD)
-    target_link_libraries(${PROJECT_NAME} INTERFACE PkgConfig::ZSTD)
+find_package(zstd REQUIRED)
+if(zstd_FOUND AND NOT SZ3_USE_BUNDLED_ZSTD)
+    target_link_libraries(${PROJECT_NAME} INTERFACE zstd::libzstd_shared)
 else ()
     add_subdirectory(tools/zstd)
     target_link_libraries(${PROJECT_NAME} INTERFACE zstd)
